Source/Scene/PostProcessorScene.js

```diff
@@ -20,23 +20,247 @@ define([
+            'uniform float u_EarthRadius; \n' +
+
+            'uniform float u_Intensity; \n' +
+
+            'varying vec2 v_textureCoordinates; \n' +
+
+            //return ndc from world coordinate biased EarthRadius
+            'vec4 GetNDCFromWC(in vec3 WC, in float EarthRadius) \n' +
+            '{ \n' +
+            ' vec4 position = vec4(WC, 1.0); \n' +
+            ' vec4 positionEC = czm_view * position; \n' +
+            ' positionEC = vec4(positionEC.x + EarthRadius , positionEC.y, positionEC.z, 1.0); \n' +
+            ' vec4 positionWC = czm_eyeToWindowCoordinates(positionEC); \n' +
+            ' return czm_viewportOrthographic * vec4(positionWC.xy, -positionWC.z, 1.0); \n' +
+            '} \n' +
+
+            //Check if current pixel is included Earth
+            //if then mask it gradually
+            'float IsinEarth(in vec2 texcoord, in vec2 SceneSize) \n' +
+            '{ \n' +
+
+                'vec2 NDC = texcoord * 2.0 - 1.0; \n' +
+                'vec4 EarthPosSC = GetNDCFromWC(vec3(0.0,0.0,0.0), 0.0); \n' +
+                'vec4 EarthPosSCEdge = GetNDCFromWC(vec3( 0.0,0.0,0.0), u_EarthRadius *1.5); \n' +
+
+                'NDC.xy -= EarthPosSC.xy; \n' +
+                'float X =  abs(NDC.x)*SceneSize.x;   \n' +
+                'X *= X;                            \n' +
+                'float Y =  abs(NDC.y)*SceneSize.y;   \n' +
+                'Y *= Y;                            \n' +
+
+                'return clamp(0.0, 1.0, max( sqrt(X + Y) / max(abs(EarthPosSCEdge.x*SceneSize.x), 1.0) - 0.8 , 0.0)); \n' +
+
+            '} \n' +
+
+
+            //For Chromatic effect
+            'vec4 textureDistorted( \n' +
+            '    in sampler2D tex, \n' +
+            '    in vec2 texcoord, \n' +
+            '    in vec2 direction, \n' +
+            '    in vec3 distortion, \n' +
+            '    in bool bSpace \n' +
+           ' ) {                    \n' +
+
+           ' vec2 SceneSize = czm_viewport.zw; \n' +
+
+            'float red = 0.0; \n' +
+            'float green = 0.0; \n' +
+            'float blue = 0.0; \n' +
+            'if(bSpace) \n' +
+            '{ \n' +
+                'red = IsinEarth(texcoord  + direction * distortion.r, SceneSize) * texture2D(tex, texcoord + direction * distortion.r).r; \n' +
+                'green = IsinEarth(texcoord + direction * distortion.g, SceneSize) *  texture2D(tex, texcoord + direction * distortion.g).g; \n' +
+                'blue = IsinEarth(texcoord  + direction * distortion.b, SceneSize) * texture2D(tex, texcoord + direction * distortion.b).b; \n' +
+            '} \n' +
+            'else \n' +
+            '{ \n' +
+                'red = texture2D(tex, texcoord + direction * distortion.r).r; \n' +
+                'green = texture2D(tex, texcoord + direction * distortion.g).g; \n' +
+                'blue = texture2D(tex, texcoord + direction * distortion.b).b; \n' +
+            '} \n' +
+
```

Comment:
Here are throughout, no need to have all this extra whitespace.  We never use more than one blank line.